File: oops/Abstract.java
package oops;

abstract class Shape {
    abstract void tell();
}

class Circle extends Shape {
    void tell() {
        System.out.println("I am a circle.");
    }
}

public class Abstract {
    public static void main(String[] args) {

        // cannot create an object of shape class, as it is abstract, the idea is to
        // keep the class as generic as possible but, at the same time, define all the
        // necessary functionality, so that it's children can be implemented wrt them.
        
        // Shape shape = new Shape();
        Circle circle = new Circle();
        circle.tell();
    }
}

File: oops/abstract.cpp
#include <bits/stdc++.h>

#define endl '\n'

using namespace std;

class Shape {
   public:
    // pure virtual function.
    virtual void draw() = 0;
};

class Circle : public Shape {
   public:
    void draw() {
        cout << "Draw a circle." << endl;
    }
};

class Rectange : public Shape {
   public:
    void draw() {
        cout << "Draw a rectange." << endl;
    }
};

int32_t main() {
    Circle circle;
    Rectange rectange;

    circle.draw();
    rectange.draw();
}